### DISCUSSION TITLE
Livewire encountered corrupt data when trying to hydrate a component

### DISCUSSION THREAD

**Discussion by discoverlance-com** [filamentphp/filament#11582] [Help]
### Package

Panel builder

### Package Version

v3.2.34

### How can we help you?

I have a resource but anytime I try to save data in the create form, I get the following error: 

Livewire encountered corrupt data when trying to hydrate a component. Ensure that the [name, id, data] of the Livewire component wasn't tampered with between requests.

I have set my APP_URL environment variable appropriately. I am not sure how to resolve this issue as there's no particular message that helps to identify where or why it's occurring.

**Comment by leandrocfe** [comment]
Does it happen in a particular resource?
Are you able to share the project on GitHub?

**Comment by discoverlance-com** [comment]
@leandrocfe 

Yes it happens on a particular resource but I don't have multiple resources to be able to confirm if it's only this resource. I am not able to share the project on github but this is how the resource is structured:

```php
<?php

namespace App\Filament\Resources;

use App\Filament\Resources\PaymentResource\Pages;
use App\Models\Payment;
use Carbon\Carbon;
use Filament\Forms\Form;
use Filament\Resources\Resource;
use Filament\Tables;
use Filament\Tables\Table;
use Filament\Forms\Components\DateTimePicker;
use Filament\Forms\Components\Select;
use Filament\Forms\Components\Textarea;
use Filament\Forms\Components\TextInput;
use Filament\Forms\Components\Wizard;
use Filament\Forms\Get;

class PaymentResource extends Resource
{
    protected static ?string $model = Payment::class;

    protected static ?string $navigationIcon = 'heroicon-o-rectangle-stack';

    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Wizard::make([
                    Wizard\Step::make('Client Details')
                        ->description('Complete user details')
                        ->icon('heroicon-m-user')
                        ->schema([
                            TextInput::make('name')
                                ->label('Full Name')
                                ->maxLength(255)
                                ->required(),
                            TextInput::make('email')
                                ->label(' Email')
                                ->email()
                                ->required(),
                            TextInput::make('phone')
                                ->label('Telephone')
                                ->tel()
                                ->telRegex('/^[+]{1}[(]{0,1}[0-9]{1,4}[)]{0,1}[-\s\.\/0-9]*$/'),
                        ]),
                    Wizard\Step::make('Billing Details')
                        ->icon('heroicon-m-credit-card')
                        ->description('Fill in billing details')
                        ->schema([
                            TextInput::make('amount')
                                ->label('Amount')
                                ->helperText('This is the amount to be requested')
                                ->prefix('GBP')
                                ->minLength(1)
                                ->numeric()
                                ->inputMode('decimal')
                                ->rules(['decimal:0,2'])
                                ->required()
                                ->dehydrateStateUsing(fn (string $state): string => (int) ($state * 100)),
                        

                        ]),
                    Wizard\Step::make('Complete')
                        ->icon('heroicon-m-clock')
                        ->description('Other details')
                        ->schema([
                            Textarea::make('comment')
                                ->helperText('Add any comments about this payment')
                                ->rows(5)
                                ->cols(5),
                        ]),
                ])
                ->columnSpanFull(),
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                //
                Tables\Columns\TextColumn::make('currency')
                    ->badge(),
               
            ])
            ->filters([
                //
                
            ])
            ->actions([
                Tables\Actions\EditAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\BulkActionGroup::make([
                    Tables\Actions\DeleteBulkAction::make(),
                ]),
            ])
            ->searchPlaceholder('Search (Created At)');
    }

    public static function getRelations(): array
    {
        return [
            //
        ];
    }

    public static function getPages(): array
    {
        return [
            'index' => Pages\ListPayment::route('/'),
            'create' => Pages\CreatePayment::route('/create'),
           // 'edit' => Pages\EditPayment::route('/{record}/edit'),
        ];
    }
}


```

**Reply by leandrocfe** [reply]
You could create another resource to check it. You could also create a new app to make sure that it is an issue.

**Reply by discoverlance-com** [reply]
Sure, the thing is I have faced the error before when I created a filament php project earlier during the first v3 release and I faced that same error. I thought it was  a config issue as I did not spend much time on that test project so I left it to test again another time. But I am facing that error again.

I am not sure if it's in relation to my coding environment, but I am using an environment similar to gitpod. So not on my local machine, I launched the project with laravel sail and I have set the APP_URL environment variable to the url the coding environment generates for me. It works alright except this issue with livewire in filamentphp. So I was assuming perhaps the urls were not matching from the livewire requests to the backend from the frontend but if it's using the APP_URL then it should have no issue. 

I have tested the form with a standalone livewire component that uses the same form from filamentphp using the advanced forms with livewire component section in the documentation and it worked with no such issues.

But I will test with another resource to see.

**Reply by discoverlance-com** [reply]
I just tried with a UserResource (using the user model) and it also failed with the same error.

![image](https://github.com/filamentphp/filament/assets/52635726/8cc6062f-d4ca-4a82-88d1-3acca1bdb103)

**Reply by AsmitNepali** [reply]
I’m experiencing the same issue as well.
<img width="1384" height="218" alt="image" src="https://github.com/user-attachments/assets/887f7d02-8e6f-472c-bc8b-aa4b663f543a" />
@discoverlance-com, were you able to find any solution for this?

**Comment by dac514** [comment]
I had this problem yesterday.

It happens when a model returns `-0` as the value _(yes, negative zero is possible in PHP floats)_

In Javascript, negative zero is **_not_** possible.

Checksum breaks because `-0 != 0`

Workaround: Find where the eloquent model is returning `-0` and normalize it, ie.

```php
// Normalize -0 to 0 while preserving other negative values
return $x == 0 ? 0 : $x;
``` 
 
Bug report in ~filament~ livewire: https://github.com/livewire/livewire/issues/2717
